client/src/components/Button/Button.js

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+
+const sizes = {
+  small: {
+    width: "30vw",
+    height: "1rem",
+    font: "Oxygen-Regular",
+  },
+  medium: {},
+  large: {
+    width: "40vw",
+    height: "2rem",
+    font: "Oxygen-Bold",
+  },
+};
+
+const SubmitButton = styled.button`
+  border-radius: 1.875rem;
+  letter-spacing: 0.07rem;
+  text-transform: uppercase;
+  color: var(--emphasis-color);
+  background: var(--basic-color);
+  font-family: ${(props) => sizes[props.size].font};
+  :active {
+    background: var(--emphasis-color);
+    color: var(--basic-color);
+  }
+  width: ${(props) => sizes[props.size].width};
+  height: ${(props) => sizes[props.size].height};
+`;
+
+/**
+ * Primary UI component for user interaction
+ */
+const Button = ({ backgroundColor, size, label, ...props }) => {
+  return <SubmitButton size={size}>{label}</SubmitButton>;
+};
```

Comment:
Maybe adjust naming as `Button` is general and returns with `SubmitButton` a "more specific" button?

Reply:
I think you don't have to return the `SubmitButton` again to pass the props, as you have defined them in the propTypes. The label could be passed as a child without specifically naming it.
https://reactjs.org/docs/composition-vs-inheritance.html#containment